projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -0,0 +1,424 @@
+        return probability_dict
+
+    def _run(self):
+        """
+        Run the circuit.
+
+        Send the circuit via the AWS Boto3 SDK. Use the provided Access Key and
+        Secret key or ask for them if not provided
+        """
+        # finally: measurements (no intermediate measurements are allowed)
+        # No explicit measurements have to be writen down. To run the circuit
+        # make implicit measurement of all the qubits
+
+        # In Braket the results for the jobs are stored in S3.
+        # You can recover the results from previous jobs using the
+        # TaskArn (self._retrieve_execution).
+        # Keept the code for retreive here to maintain the same strcuture
+        try:
+            if self._retrieve_execution is not None:
+                res = retrieve(credentials=self._credentials,
+                               taskArn=self._retrieve_execution,
+                               num_retries=self._num_retries,
+                               interval=self._interval,
+                               verbose=self._verbose)
+            else:
+                # Return if no operations are added.
+                if not self._circuit:
+                    return
+
+                n_qubit = len(self._allocated_qubits)
+                info = {}
+                info['circuit'] = self._circuithead + \
+                    self._circuit.rstrip(', ') + \
+                    self._circuittail
+                info['nq'] = n_qubit
+                info['shots'] = self._num_runs
+                info['backend'] = {'name': self.device}
+                res = send(info,
+                           device=self.device,
+                           credentials=self._credentials,
+                           s3_folder=self._s3_folder,
+                           num_retries=self._num_retries,
+                           interval=self._interval,
+                           verbose=self._verbose)
+
+            counts = res
+
+            # Determine random outcome
+            P = random.random()
+            p_sum = 0.
+            measured = ""
+            for state in counts:
+                probability = counts[state]
+                p_sum += probability
+                star = ""
+                if p_sum >= P and measured == "":
+                    measured = state
+                    star = "*"
+                self._probabilities[state] = probability
+                if self._verbose and probability > 0:
+                    print(state + " with p = " + str(probability) +
+                          star)
+
+            # register measurement result
+            for qubit_id in self._measured_ids:
+                location = self._logical_to_physical(qubit_id)
+                result = int(measured[location])
+                self.main_engine.set_measurement_result(
+                    WeakQubitRef(self.main_engine, qubit_id), result)
+            self._reset()
+        except TypeError:
```

Comment:
What part of the code in the `try` block can potentially throw a TypeError?

Reply:
That TypeError was taken from the the AQT and IBM backends at the very beginning of the coding and left behind after the changes. It seems that it is not required now

Reply:
Ok, then let's remove it.